src/Store.cs

```diff
@@ -28,32 +28,20 @@ internal void AddFuel(ulong fuel)
             }
         }
 
-        internal object? GetData()
+        internal Store Store
         {
-            var data = Native.wasmtime_context_get_data(handle);
-            if (data == IntPtr.Zero)
-            {
-                return null;
-            }
-
-            return GCHandle.FromIntPtr(data).Target;
-        }
-
-        internal void SetData(object? data)
-        {
-            var oldData = Native.wasmtime_context_get_data(handle);
-
-            var newPtr = IntPtr.Zero;
-            if (data != null)
+            get
             {
-                newPtr = (IntPtr)GCHandle.Alloc(data);
-            }
+                var data = Native.wasmtime_context_get_data(handle);
 
-            Native.wasmtime_context_set_data(handle, newPtr);
+                // Since this is a weak handle, it could be `null` if the target object (`Store`)
+                // was already collected. However, this would be an error in wasmtime-dotnet
+                // itself because the `Store` must be kept alive when this is called, and
+                // therefore this shouldn't happen.
+                var targetStore = (Store?)GCHandle.FromIntPtr(data).Target ??
+                    throw new ObjectDisposedException(nameof(Wasmtime.Store));
```

Comment:
Thinking more about this, I think the check for `null` here (and throwing in that case) probably doesn't make much sense because if the `Store` would actually have been GCed, its `Handle` might also already have been deleted, which in turn would have freed the `GCHandle`, and thus it would already be invalid (leading to UB) to try to retrieve the `GCHandle` from the `IntPtr` value in such a case.

Reply:
I removed the null check here.